synapse/python_dependencies.py

```diff
@@ -67,7 +67,7 @@
     "msgpack>=0.5.2",
     "phonenumbers>=8.2.0",
     "six>=1.10",
-    "prometheus_client>=0.0.18,<0.8.0",
+    "prometheus_client>=0.0.18",
```

Comment:
we had to pin the version because in the past, the prometheus_client developers have changed the names of the metrics without warning. If you look harder at #5636 you'll find links to https://github.com/matrix-org/synapse/issues/4001 https://github.com/prometheus/client_python/issues/317.

We can increase the version requirement here (provided you've checked that `prometheus_client` haven't introduced more backwards-incompatible changes?) but please don't remove the pin altogether.

Reply:
ah, thanks. I can confirm that synapse itself seems to run fine without issues reported by users of Arch Linux or openSUSE as far as I could see. Likely the Prometheus integration have not been used in these contexts however. Additionally I checked https://github.com/prometheus/client_python/releases/tag/v0.8.0 and found no breaking changes mentioned, differing to what https://github.com/prometheus/client_python/releases/tag/v0.4.0 describes.

I have therefore just bumped the upper, excluded version to 0.9.0 assuming that no breaking changes will be introduced in patch-versions.